src/Groups/perm.jl

```diff
@@ -387,3 +387,138 @@ function cycle_structure(g::PermGroupElem)
     # TODO: use SortedDict from DataStructures.jl ?
     return Pair{Int,Int}[ i+1 => c[i] for i in 1:length(c) if GAP.Globals.ISB_LIST(c, i) ]
 end
+
+
+# The following code implements a new way to input permutations in Julia. For example
+# it is possible to create a permutation as follow
+# pi = Oscar.Permutations.@perm (1,2,3)(4,5)(6,7,8)
+# > (1,2,3)(4,5)(6,7,8)
+# For this we use macros to modify the syntax tree of (1,2,3)(4,5)(6,7,8) such that
+# Julia can deal with the expression.
+
+
+################################################################################
+#
+#   perm
+#
+@doc Markdown.doc"""
+ @perm(ex)
+Macro to input a permutation as
+pi = @perm (1,2,3)(4,5)(6,7,8) to obtain
+(1,2,3)(4,5)(6,7,8)
+# Examples (jldoctest)
+```
+julia> @perm (1,2,3)(4,5)(6,7,8)
+(1,2,3)(4,5)(6,7,8)
+```
+"""
+macro perm(ex)
+    res = []
+   
+    while (ex).head == :call
```

Comment:
No need to write `(ex)`, just change it to `ex` everywhere.
```suggestion
    while ex.head == :call
```